cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -369,15 +369,54 @@ func (pm *PipelineManager) Configure(
 			Description:                &description,
 			ServiceManagementReference: smr,
 		}
-		servicePrincipal, err := pm.entraIdService.CreateOrUpdateServicePrincipal(
-			ctx,
-			subscriptionId,
-			spConfig.appIdOrName,
-			options)
 
-		if err != nil {
-			pm.console.StopSpinner(ctx, displayMsg, input.GetStepResultFormat(err))
-			return result, fmt.Errorf("failed to create or update service principal: %w", err)
+		var servicePrincipal *graphsdk.ServicePrincipal
+		var err error
+
+		// Loop to handle ServiceTreeNullValueError and prompt user for Service Tree ID
+		for {
+			servicePrincipal, err = pm.entraIdService.CreateOrUpdateServicePrincipal(
+				ctx,
+				subscriptionId,
+				spConfig.appIdOrName,
+				options)
+
+			if err != nil {
+				var serviceTreeError *entraid.ServiceTreeNullValueError
+				var serviceTreeInvalidError *entraid.ServiceTreeInvalidError
+				invalidInput := errors.As(err, &serviceTreeInvalidError)
+				if errors.As(err, &serviceTreeError) || invalidInput {
+					pm.console.StopSpinner(ctx, displayMsg, input.GetStepResultFormat(err))
+
+					invalidInputNotes := ""
+					if invalidInput {
+						invalidInputNotes = serviceTreeInvalidError.Error()
+					}
+
+					// Prompt user for Service Tree ID (OID)
+					serviceTreeId, promptErr := pm.promptForServiceTreeId(ctx, promptForServiceTreeIdOptions{
+						PreviousWasInvalid: invalidInputNotes,
+					})
+					if promptErr != nil {
+						pm.console.Message(ctx, output.WithErrorFormat("Failed to get Service Tree ID: %s", promptErr))
```

Comment:
Consider returning `promptErr` here? 

I believe with the current changes, a user pressing Ctrl+C ends up swallowing the error and reprompting.